src/Lecture3.hs

```diff
@@ -137,6 +169,7 @@ data List1 a = List1 a [a]
 
 -- | This should be list append.
 instance Semigroup (List1 a) where
+  (List1 a la) <> (List1 b lb) = List1 a ([b] ++ la ++ lb)
```

Comment:
That's also not the expected list append implementation 😅  It's not obvious if this is a lawful implementation because I see a compiler error in tests so the tests are not running on CI at the moment until the error is fixed 🤔 

<hr>

But I'll give some more hints on the expected behaviour.

`List1` is a type for the non-empty list. The first element of the constructor is the head of the list. So `b` is the head of the second list.

When you combine ordinary lists like `[3,1,2]` and `[5, 10]`, their concatenation is `[3,1,2,5,10]`.
You can think of `List1` (non-empty list) as a pair of the first element and the tail.

So 

* `[3, 1, 2]` becomes `(3, [1, 2])`
* `[5, 10]` becomes `(5, [10])`

And, similarly, the result of appending

* `[3,1,2,5,10]` becomes `(3, [1, 2, 5, 10])`

Your implementation with putting `b` in front will be equivalent to having `(3, [5, 1, 2, 10])` which is not a valid list concatenation.

It would be good to add a unit test that verifies that this instance indeed appends lists. You can contribute such a test to the repository itself, it would be much appreciated 🤗 Let me know if you need help with contributing or git.